accessible/field-segment.component.js

```diff
@@ -144,7 +150,12 @@ blocklyApp.FieldSegmentComponent = ng.core.Component({
   // Confirm a selection for dropdown fields.
   selectOption: function() {
     if (this.optionValue == Blockly.Msg.RENAME_VARIABLE) {
```

Comment:
I don't mean to override, but just wanted to add a data point: in the past, when I've worked with option dropdowns, we actually did compare indexes -- and then regretted it later when the indexes changed due to new items being inserted/removed in the dropdown. Comparing strings is more stable, and as a bonus it also semantically reflects the data that we care about getting from the user (i.e. we want to know that the user told us "rename!" rather than "index 2", since the latter is generally just a property of the UI element and has no further significance).

There may be other circumstances motivating this recommendation that I'm not aware about, e.g. maybe the set of possible options is set in stone, or string comparisons have too high a performance penalty, or something. But I just wanted to mention the above as something that might be worth bearing in mind.